.github/workflows/publishOnMergeWIthMain.yml

```diff
@@ -6,11 +6,26 @@ on:
   workflow_dispatch:
 
 jobs:
+  get-version:
+    name: 'Get Release Version'
+    runs-on: ubuntu-latest
+    outputs:
+      RELEASE_VERSION: ${{ steps.getMainVersion.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: 'main'
+      - id: getMainVersion
+        run: |
+          echo "version="$(node -pe "require('./packages/salesforcedx-vscode/package.json').version")"" >> $GITHUB_OUTPUT
+      - run: echo "Main Release Version is ${{ steps.getMainVersion.outputs.version }}"
   build-and-test:
     uses: ./.github/workflows/buildAndTest.yml
+    needs: [get-version]
     with: 
       branch: 'main'
+      label: ${{ needs.get-version.outputs.RELEASE_VERSION }}
   publish:
     needs: [build-and-test]
+    secrets: inherit
```

Comment:
added this so that we have the slack webhook to post during publish (note will need other secrets during publish flow)